### PR TITLE
remove cluster creation and cluster deletion so awscnfm plans can be executed

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/13458.

### CODE REVIEW

File: tekton/pipelines/aws.yaml [hunk 1]
```diff
@@ -10,29 +10,9 @@ spec:
...
-
   - name: run-tests
-    runAfter: [create-cluster]
     taskRef:
       name: aws
     workspaces:
     - name: cluster
       workspace: cluster
-
-  finally:
```
Comment (xh3b4sd) [inline tekton/pipelines/aws.yaml:32]: The idea is that`awscnfm` does the deletion. Though if `awscnfm` fails, would it make sense to have a fallback mechanism here?

Comment (MarcelMue) [reply tekton/pipelines/aws.yaml:32]: Yes a fallback would be nice. Up to you though as firecracker would have to deal with the leftovers if something goes wrong.

**Review by MarcelMue — APPROVED** [review]

**Comment by stone-z** [comment]
I know this is closed, but IMO, creating clusters in a non-standard way defeats the purpose of running conformance tests

**Comment by xh3b4sd** [comment]
Not at all, because the world is not black or white. There are tests we want to run the same way everywhere, like the upstream CNCF thingies. There is then AWS and what is relevant there. The abstracted infrastructure layer cannot and should not know about any details on how to create a cluster, because configuration to bootstrap the workload environment just tends to be very specific. The point of `awscnfm` is to follow a holistic approach so that humans can also run, debug and develop manually on a daily basis, which makes it necessary to have the whole lifecycle automation in one place.

**Comment by stone-z** [comment]
I support using `awscnfm` to create clusters for dev use but I still disagree with using it here. 
You are right, the testing platform shouldn't be concerned with cluster creation details -- it should rather use the same method which our customers will use because that is the behavior that actually matters. The `create-cluster` step uses `gsctl` directly, which is how customers create clusters.

Testing anything else is just an integration test -- which is fine, except:
This approach puts us in the same boat as with `e2e` tests where now clusters are created in non-production ways and thus will inevitably yield non-production results and require additional work to maintain parallel creation logic.

I understand `awscnfm` had to create its own clusters before `prow` existed, but I don't see the benefit of using it here.

**Comment by xh3b4sd** [comment]
> it should rather use the same method which our customers will use

`awscnfm` uses the same approach as `kubectl-gs` is doing. `gsctl` is on life support as per my understanding. So we just create CRs which get applied to the Control Plane just the same way customers would render and apply it. The only thing we wrap is different scenarious and the scheduling system is just not the right place for that. And then letting the orchestration layer only support one way to create clusters is not how our customers operate.

**Pull request merged**